src/validate.ts

```diff
@@ -0,0 +1,110 @@
+'use-strict';
+
+/**
+ * Generates random request ID 
+ * @param min Minimum number
+ * @param max Maximum number 
+ * @returns Random number between the range of the min to max number specified
+ */
+function generateRequestID(min, max){
+     return Math.floor(Math.random() * (max-min) + min).toString()
+}
+    
+/**
+ * Generates SYNC request 
+ * @returns Specified format for SYNC intent request. 
+ */
+export function generateSyncRequest(){
+    const requestId = generateRequestID(100,999);
+    return {
+        requestId,
+        inputs: [{
+          intent: "action.devices.SYNC"
+        }]
+    }
+}
+
+type CustomData = {[key: string]: any}
+
+interface QueryRequest{
```

Comment:
Nit: If we want developers to leverage all of these types we need to export them as well, correct?

Reply:
The type only needs to be exported if it's being used explicitly, ie.

```typescript
const y: QueryRequest = { ... }
```

It can be used implicitly without exporting from here or importing in another file.

```typescript
const z = generateQueryRequest(...)
console.log(z.requestId)
```